pkg/status/componentconditions.go

```diff
@@ -29,3 +32,15 @@ func NewComponentErrorCondition(reason, msg string) *toolchainv1alpha1.Condition
 		LastUpdatedTime:    &currentTime,
 	}
 }
+
+// ValidateComponentConditionReady checks whether the provided conditions signal that the component is ready, returns an error otherwise
+func ValidateComponentConditionReady(conditions ...toolchainv1alpha1.Condition) error {
```

Comment:
Please add tests for this function.

Reply:
Added in dcc42a88fe4f39f0dc61d436066a786a3c4c4215